thread/thread.h

```diff
@@ -432,7 +432,8 @@ namespace photon
     class semaphore : protected waitq
     {
     public:
-        explicit semaphore(uint64_t count = 0) : m_count(count) { }
+        explicit semaphore(uint64_t count = 0, bool in_order_resume = false)
```

Comment:
Remove explicit?

Reply:
why? if removed, an int might be unexpectedly converted to semaphore.